otus-06/src/otus_06/homework.clj

```diff
@@ -1,4 +1,43 @@
-(ns otus-06.homework)
+(ns otus-06.homework
+  (:gen-class)
+  (:require [clojure.string :as string]
+            [clojure.java.io :as io]))
+
+;; Helpers
+(defn clear-screen
+  []
+  (print (str (char 27) "[2J")) ; clear screen
+  (print (str (char 27) "[;H")) ; move cursor to the top left corner of the
+                                ; screen
+)
+
+(defn wait-for-enter
+  []
+  (let [dot-count (atom 0)
+        animation ["/" "-" "\\" "|"]]
+    (while (zero? (.available System/in))
+      (print (str "\r" "Hit Enter " (nth animation @dot-count) " "))
+      (flush)
+      (Thread/sleep 1000)
+      (swap! dot-count #(if (= % 3) 0 (inc %))))))
+
+;; Patch for (read-line) not working in terminal
+;; https://stackoverflow.com/questions/7707558/clojure-read-line-doesnt-wait-for-input
+(defn my-read-line [] (nth (line-seq (java.io.BufferedReader. *in*)) 1))
+
+
+;; Main
+(defn base-loader
+  "All in memory.
+  Instanciates via vec."
+  ([file-name schema filter-k filter-v]
+   (with-open [f (io/reader (io/resource file-name))]
+     (vec (for [line (line-seq f)
+                :let [result (zipmap schema (string/split line #"[|]"))]
+                :when (if (some? filter-k)
+                        (string/includes? (get result filter-k nil) filter-v)
```

Comment:
Just got noticed, that here might be a problem if ID is small and may intersect with others as a same sub-string. Will think about it

Reply:
Didn't get your point. You split string by separator read and then it doesn't matter is ID big or small, as I think

Reply:
For instance "3" will be caught in ids like "33" "30" and so on. Because "inclides?" will find it as a sub string.